src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2373,6 +2371,11 @@ ConnectionStateMachine(WorkerSession *session)
 				}
 
 				pollMode = PQconnectPoll(connection->pgConn);
+
+				if (status == CONNECTION_CHECK_WRITABLE && pollMode != PGRES_POLLING_OK)
+				{
+					execution->connectionSetChanged = true; /* can't break, we need the changed waitflags */
```

Comment:
CONNECTION_CHECK_WRITABLE primarily relates to target_session_attrs=read-write, which I don't think we really support right now. Do we need this check?

Reply:
we can set environment PGTARGETSESSIONATTRS or modify func NodeConninfoGucCheckHook for accept target_session_attrs to allow the cn connect the writeable worker when then nodename(pg_dist_node) have two ips。
if the first ip is read only. libpq server will close the connect and connect the next ip. libpq client receive status  CONNECTION_CHECK_WRITABLE pollMode PGRES_POLLING_WRITING and wait for the response(not receive any failed). but BuildWaitEventSet will use the PQsocket and it is closed the new socket is the next ip socket. so we should rebuild the event to get the new socket. ModifyWaitEvent is still operate the old socket what cause a epoll_ctl error.